### PR TITLE
Create the ecms_api_recipient custom module

### PR DESCRIPTION
**Pull request by pfrilling**
## Summary
This PR adds the new `ecms_api_recipient` module to the system. Installing this module will create a new user, a new role, and a new consumer that is scoped with the created user and role. A later PR will manage enabling/disabling permission for the role above.

This is setting the groundwork for a site to receive content from a hub site via the JSON API.

## Metadata

| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | yes
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | yes
| Did you perform browser testing? | yes
| Risk level | Medium
| Relevant links | [RIG-55](https://thinkoomph.jira.com/browse/rig-55)

### CODE REVIEW

File: ecms_base/modules/custom/ecms_api_recipient/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+$config['ecms_api_recipient.settings']['api_recipient_mail'] = 'automateduser@email.com';
+```
+
+### oauth_client_id
+The OAuth client id is the UUID for the consumer entity and should follow normal
+UUID length and formats.
+
+### oauth_client_secret
+The OAuth client secret is essentially the password that is used to allow
+access to the consumer. Care should be taken to ensure that this secret is
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_api_recipient/README.md:33]: Is this access to "consumer" or to the "producer?"

**Review by ben-hamelin — COMMENTED** [review]
Ok overall this looks great. I just had a minor question about the comment as it pertains to the consumer secret, and whether that was actually intended as the secret for the producer?
Otherwise we're good!

File: ecms_base/modules/custom/ecms_api_recipient/README.md (hunk 1 above)
Comment (pfrilling) [reply ecms_base/modules/custom/ecms_api_recipient/README.md]: I updated the documentation a little to hopefully be more clear. The client secret is stored with the Consumer entity that is provided by the [Consumers module](https://www.drupal.org/project/consumers). Using OAuth, publishing sites will use the id/secret to get an access token that can be used to publish content.

**Comment by ben-hamelin** [comment]
@pfrilling Ok looks good after a rebase!
/approved

**Pull request merged**